tests/unit/sagemaker/workflow/test_step_collections.py

```diff
@@ -457,7 +457,8 @@ def test_register_model_with_model_repack_with_estimator(
             assert len(request_dict["DependsOn"]) == 1
             assert request_dict["DependsOn"][0] == "TestStep"
             arguments = request_dict["Arguments"]
-            repacker_job_name = arguments["HyperParameters"]["sagemaker_job_name"]
+            assert BUCKET in arguments["HyperParameters"]["sagemaker_submit_directory"]
+            arguments["HyperParameters"].pop("sagemaker_submit_directory")
             assert ordered(arguments) == ordered(
```

Comment:
This doesn't seem related to the code change made above? Why are we popping the submit_dir?

Reply:
This test was using the `sagemaker_job_name` hyperparameter to assert on the `sagemaker_submit_directory`. Because the former hyperparameter was removed. we don't know what the `sagemaker_submit_directory` will be exactly, so do a less precise assertion on it then pop it before doing equality check on the request.

Reply:
Talked to Payton offline, we are doing necessary checks on submit dir and removing since it used job name to build the identifier, which we can't do anymore.